test/process_instances/retry_process_instance.spec.ts

```diff
@@ -0,0 +1,152 @@
+import 'mocha';
+import * as should from 'should';
+
+import {IIdentity} from '@atlas-engine/iam.contracts';
+
+import {ProcessInstanceHandler, TestFixtureProvider} from '../_setup/index';
+
+describe.only('Admin Domain: ProcessInstances.retryProcessInstance', (): void => {
+
+  let testFixtureProvider: TestFixtureProvider;
+  let processInstanceHandler: ProcessInstanceHandler;
+  let restrictedIdentity: IIdentity;
+  let superAdmin: IIdentity;
+
+  const processDefinitionIdRetryProcessInstance = 'test_admin_domain_retry_processinstance';
+  const processDefinitionIdRetryProcessInstanceFinish = 'test_admin_domain_retry_processinstance_finish';
+  const processDefinitionIdRetryProcessInstanceErrorEndEvent = 'test_admin_domain_retry_processinstance_error_endevent';
+
+  before(async (): Promise<void> => {
+    testFixtureProvider = new TestFixtureProvider();
+    await testFixtureProvider.initializeAndStart();
+
+    processInstanceHandler = new ProcessInstanceHandler(testFixtureProvider);
+
+    restrictedIdentity = testFixtureProvider.identities.restrictedUser;
+    superAdmin = testFixtureProvider.identities.superAdmin;
+
+    await testFixtureProvider.importProcessFiles([
+      processDefinitionIdRetryProcessInstance,
+      processDefinitionIdRetryProcessInstanceErrorEndEvent,
+      processDefinitionIdRetryProcessInstanceFinish]);
+  });
+
+  after(async (): Promise<void> => {
+    await testFixtureProvider.tearDown();
+  });
+
+  it('should retry a ProcessInstance and finish it', async (): Promise<void> => {
+    const result = await processInstanceHandler.startProcessInstanceAndReturnResult(processDefinitionIdRetryProcessInstance);
+    const {processInstanceId} = result;
+
+    await new Promise(async (resolve) => {
+      processInstanceHandler.waitForProcessWithInstanceIdToWalkInError(processInstanceId, async () => {
+        const processInstance = await (await testFixtureProvider
+          .adminDomainClient
+          .processInstances
+          .query({processInstanceId: processInstanceId})).processInstances[0];
+
+        should(processInstance.state).be.equal('error');
+
+        resolve();
+      });
+    });
+
+    await processInstanceHandler.wait(2000);
+
+    try {
+      await testFixtureProvider.adminDomainClient.processInstances.retryProcessInstance(processInstanceId, superAdmin);
+    } catch (error) {
+      console.error(error);
+    }
+
+    await new Promise(async (resolve) => {
+      processInstanceHandler.waitForProcessWithInstanceIdToEnd(processInstanceId, async () => {
+        const retriedProcessInstance = await (await testFixtureProvider
+          .adminDomainClient
+          .processInstances
+          .query({processInstanceId: processInstanceId})).processInstances[0];
+
+        should(retriedProcessInstance.state).be.equal('finished');
+
+        resolve();
+      });
+    });
+  });
+
+  it('should fail to retry a ProcessInstance with wrong identity', async (): Promise<void> => {
```

Comment:
```suggestion
  it('should fail to retry a ProcessInstance of another user', async (): Promise<void> => {
```

Hier wäre dann auch noch ein Test sinnvoll, der sicherstellt, dass ein SuperAdmin auch Prozessinstanzen anderer User neustarten kann.